Documentation/ApiOverview/DependencyInjection/Index.rst

```diff
@@ -9,33 +8,400 @@ Dependency injection
 ====================
 
 .. versionadded:: 10.0
-
    :doc:`Changelog/10.0/Feature-84112-SymfonyDependencyInjectionForCoreAndExtbase`
 
-TYPO3 uses a dependency injection solution based on the corresponding `PSR-11 <https://www.php-fig.org/psr/psr-11/>`_
-compliant Symfony component to standardize object initialization throughout the Core as well as in extensions.
+.. contents:: This page
+   :depth: 3
+   :local:
+
+Introduction
+============
+
+The title of this chapter is "dependency injection" (DI), but the scope is a bit
+broader: In general, this chapter is about TYPO3 object lifecycle management,
+how to obtain objects, with one sub-part of it being dependency injection.
+
+The underlying interfaces are based on the PHP-FIG standard
```

Comment:
The general structure should be more in line with the needs of readers:
- first: see how to use it quickly
- then: add explanations / implementation detail
- then: add read more section for further info
- last: add history